lib/run-compilation.js

```diff
@@ -19,6 +19,10 @@ module.exports = function runCompilation (options, compiler, done) {
     .then(function linterSuccess (lint) {
       var results = lint.results;
 
+      if (options.emitErrors === false) {
+        results = downgradeErrors(results);
+      }
+
       warnings = results.filter(function (file) {
         return !file.errored && file.warnings && file.warnings.length;
```

Comment:
I'm not sure I completely understand your comment: why would we be going through the results more than twice?

As for just leaving the errors array empty: due to the check for `!file.errored` when compiling the `warnings` array, errors wouldn't be added to the list of warnings either. We could of course expand the check to `(!file.errored && options.emitErrors !== false)`, and then another check for `options.emitErrors` before filling the `errors` array. That would make the code harder to follow I think, but if you prefer that I'd be fine with changing it.